build-boiler/build-boiler
# Eslint 3

#### Description
Bumped Eslint to 3. Also, most of our rules are abstracted into the AirB&B config (which also uses lerna 😉 under the hood). 
- you can use the `--fix` command with eslint cli to fix stuff
- are packages just use `extend` now for different environments like `web-dev` and `web-prod` instead of merging rules
- always generate the `.eslintrc` to avoid having eslint@1.x .eslintrc
- new cool plugins such as `eslint-plugin-import` to examine es6 imports/exports
I tried this in `mobile-donate` and seems to work great, adds some extra errors that we actually want it to catch. For minor fixes adding this to projects you can add a temporary `eslintrc`

```
# eslintrc-fixer.js
{
  extends: ['./node_modules/boiler-eslint-config/dist/config/web-prod.js']
}
```
And run the fix
```
eslint --fix -c eslintrc-fixer.js src/js/**
```

#### To Test
- `npm i && npm run lerna:bootstrap`
- run `gulp watch`, change some js code in `src/js` to introduce a linting error and see that it works
- do the same for build code and run `gulp lint:build`
- do the same for test code and run `gulp lint:test`

#### TODO
- check the version of webpack eslint loader if it needs a bump
- create non-react configs
- allow merging rules into hardcoded eslintrc
